components/input.tsx

```diff
@@ -0,0 +1,50 @@
+import Image from "next/image";
+import { ComponentPropsWithoutRef } from "react";
+
+interface InputProps extends ComponentPropsWithoutRef<"input"> {
+  label: string;
+  img: string;
+  error?: string;
+}
+
+export const Input = ({
+  label,
+  img,
+  error,
+  type,
+  id,
+  name,
+  ...props
+}: InputProps) => {
+  return (
+    <div className="flex flex-col gap-2">
+      <label htmlFor={id} className="text-sm font-semibold">
+        {label}
+      </label>
+      <div className="flex items-center gap-2 border-b-2 border-b-gray-500 pb-1 transition-all focus-within:border-b-red-600">
+        <label htmlFor={id}>
+          <Image src={img} alt="" width={16} height={16} />
+        </label>
```

Comment:
this couldbe be the leftAdornment expecting a react element

and just add here the {leftAdornment} then when using the component you decide what to pass there as prop